openapi-build-extension/src/main/java/io/ballerina/openapi/build/HttpServiceAnalysisTask.java

```diff
@@ -68,6 +69,11 @@
  * @since 2.0.0
  */
 public class HttpServiceAnalysisTask implements AnalysisTask<SyntaxNodeAnalysisContext> {
+    static boolean isWarningPrinted = false;
```

Comment:
This variable is to check whether the warning has printed or not because this is compiler plugin, this trigger every service node present and it will has duplicate warning message.